plugins/autochop.cpp

```diff
@@ -288,18 +292,40 @@ static bool skip_plant(const df::plant * plant, bool *restricted)
     return false;
 }
 
+static int estimate_logs(const df::plant *plant)
+{
+    //adapted from code by aljohnston112 @ github
+    df::plant_tree_tile** tiles = plant->tree_info->body;
+    df::plant_tree_tile* tilesRow;
+
+    int trunks = 0;
+    for (int i = 0; i < plant->tree_info->body_height; i++) {
+        tilesRow = tiles[i];
+        for (int j = 0; j < plant->tree_info->dim_y*plant->tree_info->dim_x; j++) {
+            trunks += tilesRow[j].bits.trunk;
+        }
+    }
+
+    return trunks;
+}
+
 static int do_chop_designation(bool chop, bool count_only, int *skipped = nullptr)
 {
     int count = 0;
+    int estimated_yield = get_log_count();
+    multimap<int, int> trees_by_size;
+
     if (skipped)
     {
         *skipped = 0;
     }
+
+    //get trees
     for (size_t i = 0; i < world->plants.all.size(); i++)
```

Comment:
Since you don't really need the index, you can simplify this to:
`for (auto plant : world->plants.all)`